server/service/schedule/schedule.go

```diff
@@ -199,6 +202,28 @@ func (s *Schedule) Start() {
 						ctxerr.Handle(s.ctx, err)
 					}
 				}
+
+				totalRunningTime := time.Since(start)
+				schedInterval = s.getSchedInterval()
+
+				if totalRunningTime > schedInterval {
+					level.Info(s.logger).Log("cron", s.name, "msg", "total runtime (%d) exceeded schedule interval (%d)")
+				}
+
+				// we need to re-synchronize this schedule instance so that the next scheduled run
+				// starts at the beginning of the next full interval
+				//
+				// for example, if the schedInterval is 1hr and the totalRunningTime is 0.2 hrs
+				// then we wait 0.8 hrs until the next time we run the schedule, or if the
+				// totalRunningTime is 1.5 hrs then we wait 0.5 hrs (skipping the scheduled run that
+				// would have overlapped with the totalRunningTime)
+				remainingWait := schedInterval - (totalRunningTime % schedInterval)
+				setWaitTimes(start, remainingWait)
+
+				clearTickerChannel(schedTicker) // in case another tick arrived during this run
```

Comment:
With the way we are using the ticker ie having to clear the channel, and reset it ... I wonder if `time.Timer` would be a better fit here. It is meant to fire once and can be reset after firing similar to `time.ticker`, but without having to drain it. This would need another change to the above

```
				ok, cancelHold := s.holdLock()
 				if !ok {
 				    // need to reset timer here
 					continue
 				}
```

`time.Ticker` has some additional complexity to account for slow receivers, which we seem to be working around here ourselves.

Open to discussion, just an idea.